### PR TITLE
LPS-150368 - FriendlyUrl resolver to calculate the alternate urls using the current logic for asset publisher

### PR DESCRIPTION
**Pull request by georgel-pop-lr**
Checked here previously: https://github.com/georgel-pop-lr/liferay-portal/pull/33

Like explained in this [comment](https://github.com/georgel-pop-lr/liferay-portal/pull/33#issuecomment-1172008847), we might have a new case surfacing if we merge this pull.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/asset/asset-display-page-api/src/main/java/com/liferay/asset/display/page/portlet/BaseAssetDisplayPageFriendlyURLResolver.java [hunk 1]
```diff
@@ -203,32 +203,42 @@ public LayoutFriendlyURLComposite getLayoutFriendlyURLComposite(
...
 
-			String urlTitle = layoutDisplayPageObjectProvider.getURLTitle(
-				locale);
+		return new LayoutFriendlyURLComposite(layout, friendlyURL, false);
+	}
 
-			if (Validator.isNotNull(urlTitle)) {
-				localizedFriendlyURL = getURLSeparator() + urlTitle;
-			}
+	protected Locale getLocale(Map<String, Object> requestContext) {
```
Comment (adolfopa) [inline modules/apps/asset/asset-display-page-api/src/main/java/com/liferay/asset/display/page/portlet/BaseAssetDisplayPageFriendlyURLResolver.java:225]: Avoid creating protected methods unless necessary; make this one private.

Also, can you avoid nesting things like this? Flat code is easier to understand than nested code. Can you do something like:
```
if (locale != null) {
    return locale;
}

....
locale = (Locale)httpSession.getAttribute(WebKeys.LOCALE);

if (locale != null) {
    return locale;
}

...
```

Comment (adolfopa) [reply modules/apps/asset/asset-display-page-api/src/main/java/com/liferay/asset/display/page/portlet/BaseAssetDisplayPageFriendlyURLResolver.java:225]: Forget about the private thing. I see it is used in subclasses.

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java [hunk 2]
```diff
@@ -8215,31 +8216,83 @@ private Map<Locale, String> _getAlternateURLs(
...
+						alternateURLSuffix =
+							layoutFriendlyURLComposite.getFriendlyURL();
+
+						break;
+					}
+					catch (PortalException portalException) {
+						if (_log.isDebugEnabled()) {
+							_log.debug(portalException);
+						}
+					}
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portal/util/PortalImpl.java:8264]: Is it safe to ignore errors here? Why not let the exception bubble up?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java (hunk 2 above)
Comment (georgel-pop-lr) [reply portal-impl/src/com/liferay/portal/util/PortalImpl.java:8264]: @robertoDiaz what do you think about this suggestion?

Comment (adolfopa) [reply portal-impl/src/com/liferay/portal/util/PortalImpl.java:8264]: Ok, it makes sense. Can you forward this PR to @liferay-echo for a final review? This changes some of their components. Thanks!

**Review by adolfopa — APPROVED** [review]

**Pull request closed without merging**